docs/upgrade/upgrade-security.asciidoc

```diff
@@ -3,75 +3,91 @@
+Problems with your {es} cluster can prevent {kib} upgrades from succeeding.
 
-IMPORTANT: The following applies to customers who are upgrading from 7.17 to 8.0 only. If you are upgrading to 8.0.1 or newer, refer to the instructions below for <<reenable-rules-upgrade, re-enabling disabled rules>>.
+During the upgrade process, {kib} creates new indices into which updated documents are written. If a cluster is approaching the low watermark, there's a high risk of {kib} not being able to create these. Reading, transforming, and writing updated documents can be memory intensive, using more available heap than during routine operation. You must make sure that enough heap is available to prevent requests from timing out or throwing errors from circuit breaker exceptions. You should also ensure that all shards are replicated and assigned.
 
-Upon upgrading to 8.0, rules are automatically disabled. Once the upgrade completes, you should re-enable them again to avoid gaps in rule coverage. _We highly recommend that you track your active rules before upgrading so you can easily find and re-enable them after upgrading to 8.0_.
+A healthy cluster has:
 
-Before upgrading, use the <<rules-api-find, Find rules API>> to capture a list of enabled rules.
+ * Enough free disk space, at least twice the amount of storage taken up by the `.kibana` and `.kibana_task_manager` indices
+ * Sufficient heap size
+ * A "green" cluster status
 
-NOTE: Use curl or another HTTP tool to run {elastic-sec} API commands.
+[float]
+==== Ensure that all {kib} instances are the same
+When you perform an upgrade migration of different {kib} versions, the migration can fail.
+Ensure that all {kib} instances are running the same version, configuration, and plugins.
 
-Here is a curl command that you can run to find enabled rules:
+[float]
+==== Back up your data
+Be sure to have a {ref}/snapshot-restore.html[snapshot] of all your data before attempting a migration.
+If something goes wrong during migration, you can restore from the snapshot and try again.
 
-[source,console]
---------------------------------------------------
-GET /api/detection_engine/rules/_find?per_page=10000&filter=alert.attributes.enabled:true
---------------------------------------------------
+Review the {kibana-ref}/resolve-migrations-failures.html[common causes of {kib} upgrade failures] and how to prevent them.
 
-To re-enable your rules from the Rules page:
 
-. Go to the Rules page (*Detect -> Rules*).
-. Select the rules that you want to enable.
-. Click *Bulk actions -> Enable* to re-enable the rules.
+[discrete]
+[[upgrade-reqs]]
+== Upgrade to an earlier 8.x version
 
 [float]
-[[reenable-rules-upgrade]]
-=== Re-enable disabled rules
-
-IMPORTANT: The following applies to customers who are upgrading from 7.17 to 8.0.1 and newer only. If you are upgrading to 8.0., refer to the instructions above for <<track-rules-upgrade, tracking rules prior to upgrading to 8.0>>.
```

Comment:
This section might still be relevant to include, since it's about "8.0.1 and newer." AFAIK, the rules would still be automatically disabled if a customer upgraded from 7.17 to 8.5.

We probably no longer need the "Important" admonition above, though, since in this context we're no longer distinguishing between `8.0.0` and `8.0.1 or newer`.